Jenkinsfile

```diff
@@ -79,6 +79,10 @@ pipeline {
             withGithubNotify(context: 'Build', tab: 'artifacts') {
               deleteDir()
               unstash 'source'
+              // prepare m2 repository with the existing dependencies
+              whenTrue(fileExists('/var/lib/jenkins/.m2/repository')) {
```

Comment:
where is that dir coming from? from the packer cache?

Reply:
that's the default home directory for the linux CI Workers, in fact, they don't have a HOME env variable but that's where all the cached artifacts are stored.

Reply:
This copies a cached folder to this build. But how does the `/var/lib/jenkins/.m2/repository` folder get populated? Wouldn't it always be empty?

Reply:
`/var/lib/jenkins/.m2/repository` gets populated every night when the daily images run the `packer_cache.sh`. It should not be empty, but it it was then it will not cause any issues at all

Reply:
got it, thanks. A minor suggestion: we could add a `du -hs .m2` so that we can verify whether this directory is non-empty.

Reply:
Do you mean temporarily? 

it was not empty when I tested it out:

see [copy step](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-java%2Fapm-agent-java-mbp%2FPR-1530/detail/PR-1530/8/pipeline/98#step-104-log-1) that took 15 seconds.

Reply:
https://github.com/elastic/apm-agent-java/pull/1530/commits/21c44bac8ebe3365b5b07be4cc3382c4be3416b1 is there now

Reply:
I guess it doesn't hurt to just stay in there